src/core/gui/XournalView.cpp

```diff
@@ -300,11 +301,11 @@ auto XournalView::onKeyPressEvent(GdkEventKey* event) -> bool {
 
     // Switch color on number key
     auto& colors = control->getWindow()->getToolMenuHandler()->getColorToolItems();
-    if (!state && (keyval >= GDK_KEY_0) && (keyval < GDK_KEY_0 + std::min((std::size_t)10, colors.size()))) {
+    if ((keyval >= GDK_KEY_0) && (keyval < GDK_KEY_0 + std::min((std::size_t)10, colors.size()))) {
```

Comment:
Remove this change. <Control>+number should switch between drawing types and colors should only be switched, when no modifier is pressed.

Reply:
Maybe then we need to mask out consumed modifiers like in `TextEditor::onKeyPressEvent`
>
    /*
    According to https://docs.gtk.org/gdk3/method.Keymap.translate_keyboard_state.html
    consumed modifiers should be masked out. For instance, on a US keyboard, the plus symbol is shifted, so when
    comparing a key press to a <Control>plus accelerator <Shift> should be masked out.
    */

The Ctrl+NUMBER accelerator is set in `main.glade`, e.g.
```glade
                            <signal name="toggled" handler="ACTION_TOOL_DRAW_SPLINE" swapped="no"/>
                            <accelerator key="8" signal="activate" modifiers="GDK_CONTROL_MASK"/>
```